From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas McKahan <tonymckahan@gmail.com>
Date: Sat, 6 Oct 2018 22:50:14 -0400
Subject: Odroid C2 enable SPI

Odroid C2 enable SPI

- f928b31d8a1983fd8cfd9c97de084e532283b106: 1543550719: Thomas McKahan <tonymckahan@gmail.com>: '[ meson64-dev ] fix Odroid C2 boot, add spidev'
---
 arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts | 26 ++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts b/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
index b2cb12fb46fd..c252de8e4b17 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
@@ -176,6 +176,32 @@ hdmi_connector_in: endpoint {
 		};
 	};
 
+	spi-gpio {
+		compatible = "spi-gpio";
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+		ranges;
+		status = "disabled";
+		sck-gpios  = <&gpio GPIOX_2 0>;
+		miso-gpios = <&gpio GPIOX_4 0>;
+		mosi-gpios = <&gpio GPIOX_7 0>;
+		cs-gpios  = <&gpio GPIOX_3 0
+			     &gpio GPIOX_1 0>;
+		num-chipselects = <2>;
+
+		/* clients */
+		spidev0@0 {
+			compatible = "spidev";
+			reg = <0>;
+			spi-max-frequency = <500000>;
+		};
+		spidev0@1 {
+			compatible = "spidev";
+			reg = <1>;
+			spi-max-frequency = <500000>;
+		};
+	};
+
 	sound {
 		compatible = "amlogic,gx-sound-card";
 		model = "ODROID-C2";
-- 
Armbian

